usr/lib/common/obj_mgr.c

```diff
@@ -732,6 +739,13 @@ CK_RV object_mgr_destroy_object(STDLL_TokData_t *tokdata,
         return CKR_OBJECT_HANDLE_INVALID;
     }
 
+    if (!object_is_destroyable(o)) {
+        TRACE_ERROR("Object is not destroyable\n");
+        object_put(tokdata, o, TRUE);
+        o = NULL;
```

Comment:
I know, but since object_put decreases the ref count of the object, and the object might have been deleted if the ref count has reached zero, I use this as an indication that o should not be accessed any more.  You will find this an many places....